lib-js/interactor.ts

```diff
@@ -1,207 +1,207 @@
-      result.push(option)
+      result.push(option);
     }
   }
-  return result
+  return result;
 }
 
 export class Interactor {
-  private memory: Record<string, InteractorOption> = {}
-  private input: NodeJS.ReadableStream
-  private output: NodeJS.WritableStream
+  private memory: Record<string, InteractorOption> = {};
+  private input: NodeJS.ReadableStream;
+  private output: NodeJS.WritableStream;
 
   constructor(input: NodeJS.ReadableStream, output: NodeJS.WritableStream) {
-    this.input = input
-    this.output = output
+    this.input = input;
+    this.output = output;
   }
 
-  private printLine(line: string = ""): void {
-    this.output.write(`${line}\n`)
+  private printLine(line = ''): void {
+    this.output.write(`${line}\n`);
   }
 
   private printOptions(options: InteractorOption[], test: TestCase): void {
-    for (const { key, description } of options) {
+    for (const {key, description} of options) {
       const _description =
-        typeof description === "string" ? description : description(test)
-      this.output.write(`${key}. ${_description}\n`)
+        typeof description === 'string' ? description : description(test);
+      this.output.write(`${key}. ${_description}\n`);
     }
   }
 
   // Prints content bounded by a delimiter
   private printContent(content: string): void {
-    const width = Math.max(...content.split("\n").map((l) => l.length))
-    const delimiter = Array(width).fill("*").join("")
-    this.printLine()
-    this.printLine(delimiter)
-    this.printLine(content)
-    this.printLine(delimiter)
-    this.printLine()
+    const width = Math.max(...content.split('\n').map(l => l.length));
+    const delimiter = Array(width).fill('*').join('');
+    this.printLine();
+    this.printLine(delimiter);
+    this.printLine(content);
+    this.printLine(delimiter);
+    this.printLine();
   }
 
   /**
    * Run the interactor prompt on the given test case.
    */
   async prompt(test: TestCase): Promise<void> {
-    const rl = readline.createInterface(this.input, this.output)
+    const rl = readline.createInterface(this.input, this.output);
 
     function question(prompt: string): Promise<string> {
-      return new Promise((resolve) => {
-        rl.question(prompt, resolve)
-      })
+      return new Promise(resolve => {
+        rl.question(prompt, resolve);
+      });
     }
 
-    const type = test.result().failureType || ""
+    const type = test.result().failureType || '';
 
     // If a repeated choice is chosen for a given failure type, run that choice
     if (this.memory[type]) {
-      const choice = this.memory[type]
-      await choice.resolve(test)
-      rl.close()
-      return
+      const choice = this.memory[type];
+      await choice.resolve(test);
+      rl.close();
+      return;
     }
 
-    while (true) {
-      this.printLine()
-      this.printLine(`In test case: ${test.dir.relPath()}`)
-      this.printLine(test.result().message)
+    for (;;) {
```

Comment:
Is there a reason why gts prefers this? I can't find any mention of it in the style guide, so I wonder if this is a bug

Reply:
See https://eslint.org/docs/rules/no-constant-condition. I think it's weird that they specifically choose to disallow `while (true)`, since that's *not* likely to be a bug, but it's definitely intentional.